tests/suites/spaces_ec2/juju_bind.sh

```diff
@@ -15,7 +15,10 @@ run_juju_bind() {
 	hotplug_nic_id=$2
 	add_multi_nic_machine "$hotplug_nic_id"
 	juju_machine_id=$(juju show-machine --format json | jq -r '.["machines"] | keys[0]')
-	ifaces=$(juju ssh ${juju_machine_id} 'ip -j link' | jq -r '.[].ifname | select(. | startswith("enp") or startswith("ens"))')
+	wait_for_machine_netif_count "$juju_machine_id" "2"
```

Comment:
I don't think this is what we're looking for. This will pass immediately since by default juju machines have 2 network interfaces, the `ens` interface, and the `fan` address

Unfortunately our `wait` helper functions wait on aspects of juju, so can't really be applied here

You'll want something like:
```
while [ !  $(echo $ifaces | wc -l) -eq 2]
   sleep 1
   ifaces=$(ip -j link | jq -r ".[].ifname")
done
```

Reply:
you are absolutely right, I fixed it a little bit:

```sh
	while [[ ! $(echo $ifaces) -eq 2 ]]
	do
   		sleep 1
		ifaces=$(ls /sys/class/net | wc -l)
	done
```

Reply:
I ended up adding 
```sh
	# Wait until the new NIC is UP
	juju ssh ${juju_machine_id} 'while [[ ! $(echo $ifaces) -eq 2 ]]
	do
		sleep 1
		ifaces=$(ls /sys/class/net | grep "ens\|enp\|eth" | wc -l)
	done'
```
on the `add_multi_nic_machine()` instead of twice in juju_bind.sh and upgrade_charm_with_bind.sh